stl/inc/ranges

```diff
@@ -1454,20 +1457,101 @@ namespace ranges {
     template <class _Rng>
     inline constexpr bool enable_borrowed_range<ref_view<_Rng>> = true;
 
+    // clang-format off
+    template <range _Rng>
+        requires (movable<_Rng> && !_Is_initializer_list<_Rng>)
+    class owning_view : public view_interface<owning_view<_Rng>> {
+        // clang-format on
+    private:
+        _Rng _Range{};
+
+    public:
+        // clang-format off
+        owning_view() requires default_initializable<_Rng> = default;
+        // clang-format on
+
+        constexpr owning_view(_Rng&& _Range_) noexcept(is_nothrow_move_constructible_v<_Rng>) // strengthened
+            : _Range(_STD move(_Range_)) {}
+
+        owning_view(owning_view&&) = default;
+        owning_view& operator=(owning_view&&) = default;
+
+        _NODISCARD constexpr _Rng& base() & noexcept {
+            return _Range;
+        }
+        _NODISCARD constexpr const _Rng& base() const& noexcept {
+            return _Range;
+        }
+        _NODISCARD constexpr _Rng&& base() && noexcept {
+            return _STD move(_Range);
+        }
+        _NODISCARD constexpr const _Rng&& base() const&& noexcept {
+            return _STD move(_Range);
+        }
+
+        _NODISCARD constexpr iterator_t<_Rng> begin() noexcept(noexcept(_RANGES begin(_Range))) /* strengthened */ {
+            return _RANGES begin(_Range);
+        }
+
+        _NODISCARD constexpr sentinel_t<_Rng> end() noexcept(noexcept(_RANGES end(_Range))) /* strengthened */ {
+            return _RANGES end(_Range);
+        }
+
+        _NODISCARD constexpr auto begin() const
+            noexcept(noexcept(_RANGES begin(_Range))) /* strengthened */ requires range<const _Rng> {
+            return _RANGES begin(_Range);
+        }
+
+        _NODISCARD constexpr auto end() const
+            noexcept(noexcept(_RANGES end(_Range))) /* strengthened */ requires range<const _Rng> {
+            return _RANGES end(_Range);
+        }
+
+        _NODISCARD constexpr bool empty() noexcept(noexcept(_RANGES empty(_Range))) /* strengthened */
+            requires _Can_empty<_Rng> {
+            return _RANGES empty(_Range);
+        }
+        _NODISCARD constexpr bool empty() const noexcept(noexcept(_RANGES empty(_Range))) /* strengthened */
+            requires _Can_empty<const _Rng> {

```

Comment:
Can't we write the requires requires constraint directly? or does that need requires expression support that we don't have?

Reply:
`view_interface` has four occurrences of this same ad-hoc constraint, so I made a helper concept for DRY purposes. @miscco implemented `ref_view` very shortly thereafter and reused the helper, I assume also just for DRY. I copy-pasted `ref_view` to make `owning_view` and didn't give this constraint any thought at all.

TLDR: Everybody hates `requires requires`. Shout if you'd prefer that we spell it out explicitly here (and/or elsewhere) so as to more obviously agree with the Standard wording. Both are equivalent under as-if.

Reply:
I'm happy with the named concept.